### PR TITLE
Validate email on blur

### LINKED ISSUE #38: Email address validity should be checked on blur
#### Describe the bug

Login screen: the email address (1) is checked on change and (2) does not display the error message, but instead only disables the login button.

Same with the reset password screen

The create an account page is doing the right thing with the error message, but the email address is still checked on change.

#### What is the expected behavior?

Email address validity (whether the email entered is of the right regex format) should be checked on blur.

You can still disable the login button, and do an on change check to enable the button when both the required fields are not empty. 

#### What are the steps to reproduce?
1. spins up the iOS example project
2. type invalid email addresses in the aforementioned screens.

#### Screenshots or links to minimum reproduction example
image

### PR DESCRIPTION
**Pull request by daileytj**
Fixes #38.

#### Changes proposed in this Pull Request:
- Validate email inputs on blur

### CODE REVIEW

**Comment by daileytj** [comment]
Putting this up as a draft for now. 

I'm stuck on two things here:
- I can't figure out how to clean up the linting errors.
- The create account page's email input validation seems to be off by 1 character so a user can enter a valid email to enable the next button and delete 1 character making the email invalid but can still progress.

**Review by joebochill — CHANGES_REQUESTED** [review]
I think we can simplify this logic a little bit. Instead of having two useState values, we can just use a single boolean:
```
const [hasEmailFormatError, setHasEmailFormatError] = React.useState(false);
```

And then set it onBlur:
```
onBlur={(): void => {
    if(emailInput.length > 0 && !EMAIL_REGEX.test(emailInput)) setHasEmailFormatError(true);
}}
```

and unset it either onChange or onFocus (whichever is the desired behavior):
```
onChangeText={(text: string): void => {
    setEmailInput(text);
    setHasEmailFormatError(false);
}}
```

**Comment by joebochill** [comment]
That should get you around any linting issues too.

File: login-workflow/src/subScreens/CreateAccount.tsx [hunk 1]
```diff
@@ -83,17 +84,22 @@ function isValidEmail(text: string): boolean {
...
-    };
 
-    const showEmailError = emailInput.length !== 0 && !isValidEmail(emailInput);
+    const validateEmail = (text?: string): void => {
+        if (text ? !isValidEmail(text) : !isValidEmail(emailInput)) {
+            setEmailError(t('MESSAGES.EMAIL_ENTRY_ERROR'));
+            props.onEmailChanged('');
+        } else {
+            setEmailError('');
+            text ? props.onEmailChanged(text) : props.onEmailChanged(emailInput);
```
Comment (joebochill) [inline login-workflow/src/subScreens/CreateAccount.tsx:100]: This is where the linting issue is. The ternary operator is usually used for variable assignment, not function calls like this. I would just use a simple if statement here:
```suggestion
            if( text){ props.onEmailChanged(text); }
            else { props.onEmailChanged(emailInput); }
```

**Review by joebochill — APPROVED** [review]

**Pull request merged**